### ISSUE TITLE
'Choose what happens before' broken in DisplayMagician v2.1.2

### ISSUE THREAD

**Issue by erbkaiser** [terrymacdonald/DisplayMagician#67]
**Describe the bug**
After upgrading from DisplayMagician v2.0.1 to v2.1.2 the 'Choose what happens before' option is broken. It is not possible to add a new program and if  there is/are existing entries, the first one is partially under the program GUI and cannot be edited.

**To Reproduce**
Steps to reproduce the behaviour:
Existing:
1. Downgrade to 2.0.1 and create any 'start programs' for a game shortcut
2. Upgrade to 2.1.2 and edit the shortcut
3. Try to edit existing entries

New:
1. Have 2.1.2 installed and go to 'Choose what happens before' for a game shortcut
2. Click 'Add Start Program'
3. See nothing happen

**Expected behaviour**
The ability to add new start programs and to edit existing entries.

**Screenshots**
![image](https://user-images.githubusercontent.com/6014923/146675320-e51c212b-9b51-4641-b7dd-73538d123a0e.png)

**Reporting Logs**
If DisplayMagician starts:

[DisplayMagician-Support-20211219-1346.zip](https://github.com/terrymacdonald/DisplayMagician/files/7740806/DisplayMagician-Support-20211219-1346.zip)

**Enviroment (please complete the following information):**
 - Windows Version: Win11
 - DisplayMagician Version: 2.1.2
 - Video Card: nVidia Gigabyte RTX 2060 Super
 - Number and make/model of monitors: 3x Samsung Syncmaster 
 - Date of last video driver update: 1 Dec 2021
 - Date of last windows update: 19 Dec 2021

**Additional context**

**Comment by terrymacdonald** [comment]
Ah shoot! I'll go look at it right away.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Confirmed bug. Forgot to reassign the old click events after adjusting the document object hierarchy within the page. Now fixed in codebase. Will be released in v2.1.3.

**Comment by terrymacdonald** [comment]
Closing issue so I can keep track of other issues. Feel free to reopen if the next release doesn't fix the issue for you.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@erbkaiser Can you please test this new test version of DisplayMagician to make sure the problem is now fixed?

Test DisplayMagician v2.1.2.6:
[DisplayMagicianSetup-v2.1.2.6.zip](https://github.com/terrymacdonald/DisplayMagician/files/7742390/DisplayMagicianSetup-v2.1.2.6.zip)
 
Thanks
Terry

**Comment by erbkaiser** [comment]
Confirmed fixed!

**Comment by terrymacdonald** [comment]
@erbkaiser I need someone I know well to test out this new version that is just about ready for release

DisplayMagician was crashing as  had introduced a bug in 2.x that would incorrectly add all the non-steamGames to the list of SteamGames. This was not a valid thing to do. It would crash DisplayMagician and would have affected anyone with Steam and any one more GameLibrary (e.g. GOG, Epic, Uplay etc).

I also managed to find a potential issue with editing shortcuts that I've hopefully now averted. We now properly 
* Properly clear out the form and all it's storage variables when the form loads (done)
* Add in some protection logic to avoid a crash even if this situation occurs again (partly done)

Additionally, I found a recursive logic flaw that meant that DisplayMagician used WAAAAAAAAAAAAAY more memory than it needed to. I'm talking 5GB of memory rather than the 129MB it was supposed to do. 

So please test this out for me ASAP so I can drop a release 'tout de suite'.  I think this issue could be hurting a few people :(.

[DisplayMagicianSetup-v2.1.3.10.zip](https://github.com/terrymacdonald/DisplayMagician/files/7766758/DisplayMagicianSetup-v2.1.3.10.zip)

Thanks
Terry

**Comment by erbkaiser** [comment]
I saw DM was not running before updating so probably I was getting the same crash. However, with the new 2.1.3, DisplayMagician crashes when I try to create or edit a game shortcut
[shortcut crash.zip](https://github.com/terrymacdonald/DisplayMagician/files/7768219/shortcut.crash.zip)
.

**Comment by erbkaiser** [comment]
2.1.4.1 (pre-release) has the same crash, while DisplayMagician is extracting icons it disappears.

**Comment by terrymacdonald** [comment]
Hmmmm. Ok. I take a look tomorrow morning as soon as I can.

Thanks for letting me know.

Terry

**Comment by terrymacdonald** [comment]
@erbkaiser I've now got a new DisplayMagician to test!

I've completely rewritten the Shortcut Editing Form logic, and it's WAY more logical and robust. I've also taken the time to add a couple of new features because... well why not?

Features:
- Can run Start Programs as Administrator
- Can run Main Application as administrator
- Can run Stop Program as Administrator (the run program afterwards one)
- Many robustness improvements
- Copy Shortcut works properly
- Made UI selection clearer in a couple of places

Can you please test 2.1.4.67 which I've attached to this issue. It is nearly ready to go out as a release, so I'd really appreciate it if you could test the Shortcut Editing Form for me.

[DisplayMagicianSetup-v2.1.4.67.zip](https://github.com/terrymacdonald/DisplayMagician/files/7776459/DisplayMagicianSetup-v2.1.4.67.zip)

Thanks
Terry

**Comment by erbkaiser** [comment]
Hi Terry,

the new build looks great. Launching programs before and after works great and the administrator function also kicks in properly. Both existing and new shortcuts function like they should.

thanks for working on this over the holiday!

**Comment by terrymacdonald** [comment]
Hi @erbkaiser 

Yet another favour to ask. Can you please test this new test build for me? I found a bug in the StopProcess function that was preventing the stop from working in some situations. If the program being stopped didnt have a window associated with it, it probably wouldn't be stopped. This has now been fixed in the codebase.

I have also managed to implement some extra cool multithreading/single instance/named pipes coolness that should speed up DisplayMagician being used in the system tray. It introduces a lot of new multi-threading and there is a real opportunity for this causing lock-ups and hangs. I was wondering if you would be able to test this build for me please?

I've included a test build for you to install and try. Please try and break it as I need to ensure that this new code is robust.
[DisplayMagicianSetup-v2.1.5.33.zip](https://github.com/terrymacdonald/DisplayMagician/files/7806772/DisplayMagicianSetup-v2.1.5.33.zip)

Thanks
Terry

**Comment by erbkaiser** [comment]
Hi Terry,

this build broke display profile switching from the program and system tray for me, it does nothing if I select a different profile than I am currently using. It will still switch display profiles from game shortcuts.

**Comment by terrymacdonald** [comment]
Ah thanks. I think I've missed connecting those methods into the new
multitasking process. They're probably using the original launching method.

I'll take a closer look tomorrow. Thanks for testing!

Terry

**Comment by terrymacdonald** [comment]
@erbkaiser OK. I fixed those bugs as well as I can tell. I missed a couple of commands out, and had a logic error when checking for a background task being run, and the tasks were in non-blocking mode when I need them in blocking mode to stay compatible with the UI (they'll be non-blocking when we move to .net6 and MAUI later in the year.

I also found another bug with the stop program that's run afterwards. That wasn't quite working, but should now be fixed!

Can you please test this one out for me as well? I really can't thank you enough for all your thorough testing. 🏅 
[DisplayMagicianSetup-v2.1.5.43.zip](https://github.com/terrymacdonald/DisplayMagician/files/7813659/DisplayMagicianSetup-v2.1.5.43.zip)

Oh, can you please confirm that if the installation process goes smoothly too when you install? I'm trying a slightly different way to handle restarting explorer to get the desktop background menu to install more reliably.

Thanks
Terry

**Comment by erbkaiser** [comment]
On my Win11 system, the installer tried to close explorer.exe and failed so I kept re-trying, on the third attempt it succeeded. After the install was complete it didn't start explorer.exe so I was without a task bar and desktop. I had to relaunch explorer.exe through task manager.
I forgot to mention it with the previous build, but there it also gave me a warning twice, but there at the end of the installation explorer.exe reopened.

I have the DisplayMagician profiles in the extended right-click menu on desktop and they are working. The menu was missing for me since upgrading to Win11 so that is now working better.

On to the tests.

The name of the systray icon is not update correctly:
After changing from 'Three Screens' to 'Surround' the tray icon still says 'DisplayMagician (Three Screens)' but 'Surround' is shown in boldface (active). I can change back to 3s with any method though.
This happens when changing from the systray as well as from the desktop background menu.
If I open the DM program itself, it properly identifies it is currently in the 'Surround' profile.

Changing display profiles using the program lead to a crash after the profile changed, DM fully closed. Attached is the log file from directly after the crash, prior to restarting DM. [crash.zip](https://github.com/terrymacdonald/DisplayMagician/files/7813868/crash.zip)
This is not repeatable and happened only once so far.

Game launching from either the program or a desktop shortcut works fine. Any additional programs opened that should close also close, although I don't have any headless programs (no window) to test with.

**Comment by terrymacdonald** [comment]
OK @erbkaiser, I've made a few changes to the way that DisplayMagician works in the hope that it fixes most of the issues you have found. Please test this new version: 
[DisplayMagicianSetup-V2.1.5.50.zip](https://github.com/terrymacdonald/DisplayMagician/files/7820559/DisplayMagicianSetup-V2.1.5.50.zip)

I've rolled back the DisplayMagician Installer changes that I experimented with. I have tried a slightly different mode for the restart manager in the hope that it will restart explorer.exe if it ever needs to be restarted as part of an upgrade. This is only ever required if the user has previously used the DisplayMagician Desktop Background context menu since the PC was started, as using the desktop background context menu loads the DLL into the explorer.exe process. This means that the uninstaller needs to tell the Windows Restart Manager to restart explorer.exe in order free up the DLL so it can be deleted. 

Unfortunately Restart Manager can't always restart explorer.exe it seems, and that is when the start button and taskbar disappears.... when Restart Manager can't restart explorer.exe. To combat the rare occasions this occurs, I've added a 'Do you want to restart now?'  prompt to the end of the installation. This will hopefully give people the opportunity to restart their machines even if explorer.exe doesn't restart, so they can at least get their PCs back up an running. 

I really wish there was something else I could do about this fact, as it's crap this keeps happening. I'm contemplating rewriting the installer using InstallShield as it is really beginning to annoy me. I've worked so hard to make the rest of the process so reliable, and it sucks when the installer is the one thing I have the most problem with.

The systray icon should update with the messages correctly now. As I moved DM to a 'single instance' model (where a single instance is loaded, even if you attempt to run a second copy it will close down and pass the information to the first instance), I also had to develop a threaded processing model for the ChangeProfile and RunShortcut tasks to get them to work when passed the command line information from instances running from desktopshortcuts and the desktop context menu. This change also had another effect of meaning I needed a thread safe way of passing information back to the main instance to update the systray icon.  It got complicated real quick :)

So, the systray icon should now properly update the systray icon without crashing DisplayMagician :crossedfingers:

That brings me to the final issue of the crash you had. What NVIDIA drivers are you running now? I had a crash last night while running a game without using DisplayMagician, and I'm wondering if there is an issue with the NVIDIA drivers? It would be weird if that's true, but the fact that I had the problem without running DisplayMagician makes me think something weird is up.... Anyway - I have v497.29,

I checked your crash.zip and it all looks fine. It looks like it crashed just after it was trying to get the topology from the mosaic cards:

```
2022-01-05 10:52:42.6878|TRACE|DisplayMagician.Program|NVIDIALibrary/GetNVIDIADisplayConfig: NvAPI_EnumPhysicalGPUs returned 1 Physical GPUs
2022-01-05 10:52:42.6878|TRACE|DisplayMagician.Program|NVIDIALibrary/GetNVIDIADisplayConfig: NVIDIA Video Card is one from the GeForce range
2022-01-05 10:52:42.7978|TRACE|DisplayMagician.Program|NVIDIALibrary/GetNVIDIADisplayConfig: NvAPI_Mosaic_GetCurrentTopo returned OK.
2022-01-05 10:52:42.7978|TRACE|DisplayMagician.Program|NVIDIALibrary/GetNVIDIADisplayConfig: NVIDIA Mosaic is enabled.
2022-01-05 10:52:42.7978|TRACE|DisplayMagician.Program|NVIDIALibrary/GetNVIDIADisplayConfig: NvAPI_Mosaic_GetTopoGroup returned OK.
```

The missing log means it crashed at this line of code:

`if (mosaicTopoBrief.Topo != NV_MOSAIC_TOPO.TOPO_NONE && mosaicTopoBrief.Enabled == 1)`

It could be that the structure returned by the NVAPI driver is sometimes corrupted, so it crashes when attempting to access that information? Something to test possibly We'll see how this testing goes.

Thanks for doing this testing for me
It really does help speed things up.

Terry

**Comment by erbkaiser** [comment]
I'm on the same nVidia driver version. Right now there is no newer one available for my hardware. I haven't encountered any other crashes, but then again I haven't been gaming that much other than the truck sim games which I launch through DM.

The install went well this time, properly restoring explorer.exe but it did show the request to reboot, which I denied this time. The desktop background menu is still there and works.

The systray updates properly after switching.

Game shortcuts also work properly.

Glad to be able to help out with testing :)

**Comment by terrymacdonald** [comment]
Nice! OK - please keep an eye on it and tell me if there is anything that causes an error with that version.

I have a question about the reboot prompt at the end. Do you think it is too much? Will it confuse people? 

Thanks
Terry

**Comment by erbkaiser** [comment]
I think the reboot prompt is a good addition. It's not out of place, more programs do this, and it would help if something goes wrong during the install.

**Comment by terrymacdonald** [comment]
OK @erbkaiser - I've got a new version to test which I hope has all of the errors removed from it. Can you please do the normal amazing testing that you do and report back if you find any errors please?

Please report back with any errors you find (screenshots too please if possible) and of course upload the support ZIP file if you have any errors too.

[DisplayMagicianSetup-v2.2.0.41.zip](https://github.com/terrymacdonald/DisplayMagician/files/7866290/DisplayMagicianSetup-v2.2.0.41.zip)

Thanks (as always)
Terry

**Comment by erbkaiser** [comment]
Took me a bit to do the tests.

Installation worked fine, it properly closed the existing DisplayMagician and prompted me to reboot at the end.

At first glance, all features work. Using and creating display profiles work, as does switching between them.
Game shortcuts also work for the most part.

Starting this build however, they stopped executing two 'choose what happens before' programs for my truck sim games: 
![image](https://user-images.githubusercontent.com/6014923/149621128-fc082dc2-b7f7-43b0-b914-9d2cc66282b8.png)
The TrackIR5 program has been problematic for longer, but the Steam shortcut (to force Steam into Big Picture mode) 100% worked reliably until this build.

If I add for example Notepad to the list, that WILL launch.

Support zip attached, there are multiple game launches in here.
[DisplayMagician-Support-20220115-1318.zip](https://github.com/terrymacdonald/DisplayMagician/files/7874928/DisplayMagician-Support-20220115-1318.zip)


I am also experiencing a rarely occurring weird issue immediately after swapping display profiles sometimes where the entire Windows system seems to lock up and it will beep incessantly. The system is so locked up when this happens that I can't even get task manager to open, and the mouse moves with a delay. This happens for up to 3 minutes and then it is fine again. I have been completely unable to find any cause or someone online talking about this.
I am still trying to track this down and will try to make a video next time it happens. Of course, when trying to trigger it on demand, it would not happen. It might be in the support.zip attached, though. It happened after launching ETS2 in Surround mode, quitting the game, and Windows returning to the Three Screens profile.

**Comment by erbkaiser** [comment]
I've been able to track the beeping down to wallpaper changing. As this is handled on my machine by the DisplayFusion app I think it's related to that, and not DisplayMagician.

**Comment by terrymacdonald** [comment]
Ok, that's interesting. DisplayMagician word to register to make the
desktop background changes. Do all of your shortcuts have 'Do nothing'
selected?

I'm hoping to do some more trackIR start program checks later this week if
I can, so I can figure out why is not starting for you. I bet it's
something silly on my end.

I'll keep you up to date.

Also, do you mind sending me your email address to
***@***.*** so we don't end up communicating through a
really old GitHub issue? Id really like to be able to send you build as
soon as you get them as you're the best tester I've ever met (and that
includes professionally in my day job!).

Thanks
Terry

**Comment by erbkaiser** [comment]
I don't know what you mean with 'do nothing' here.

Seems Github doesn't like emails. You can reach me at jordi àt erbkaiser dót nl
![image](https://user-images.githubusercontent.com/6014923/149733979-1620d93d-46da-4111-be54-9721011ab66f.png)

Cheers

**Comment by terrymacdonald** [comment]
Oh sorry. I was meaning that if you select each profile, then click on
Profile Settings button, then check wallpaper is set to 'Do nothing'.

Sorry for not being clear. Sometimes I'm already thinking about the next
problem while typing the answer to you!

Also, I just fixed the TrackIR start program issue. Turns out that TrackIR
needs admin rights to run. I've managed to build some auto detection into
DisplayMagician that will automatically request admin rights if windows
says that the program can't run as it requires elevated privileges. I've
tested that with TrackIR and it works!

This might also work for Steam, but I have to tell you that I don't support
running Steam as a start program. It may affect the running of Steam games
(if they are in the same shortcut), so I tell people to remove the Steam
start programs when I see them. But if it works then I guess leave it?

And thanks for your email. Are you ok with being the first person to get a
release to test each time I'm ready to release one? I don't want to give
you to much work to do.

Thanks
Terry

**Comment by erbkaiser** [comment]
Hi Terry,

I'm working from home most of the time so I usually have enough time to test new builds. I'll gladly do this.

Display profiles: Got it, and yes all display profiles have 'Do nothing selected'.

Great news on the TrackIR fix and that it helps with admin rights detection in general!

The Steam shortcut is just a convenience thing since I want to use the Steam controller API for some games, and unfortunately that is only available if Steam runs in Big Picture mode. Passing the pseudo-URL steam://open/bigpicture to Steam forces it to relaunch in this mode.
DisplayMagician doesn't support URLs so I use the Steam exe to get it to work directly.
I don't like Steam Big Picture mode in general, so I only enable it when I want to run a game I plan to play with a controller.

**Comment by erbkaiser** [comment]
I have now been able to catch a video of the beeping hell issue. https://youtu.be/arnE0I4tHsQ
Video should be live in ~5 minutes. 
What you see happening there is Windows freaking out after it has changed back to 'Surround' from 'Three Screens'. The beeps are coming from my speakers. As you can also see my mouse is incredibly delayed, I am trying to make smooth movements. Once the GoG Galaxy screen pops up (unrelated issue) you see me fruitlessly try to get the mouse to the close button.

This started when I was launching Euro Truck Simulator 2 from the desktop game shortcut which should put it in the Surround profile. After switching to Surround the beeping hell started, and lasted until DisplayMagician restored my original Three Screens profile.
ETS2 launched _after_ that, in the wrong display profile.
Support zip: 
[DisplayMagician-Support-20220117-1443.zip](https://github.com/terrymacdonald/DisplayMagician/files/7882212/DisplayMagician-Support-20220117-1443.zip)

**Comment by terrymacdonald** [comment]
Hmmmmm. Those are some very weird logs you gave me @erbkaiser!

It looks to me like the NVIDIA driver is having a massive slowdown for some reason (or windows). There are some really weird errors in the log files too, when DisplayMagician is trying to get the current windows display config from windows. I've got a selection of them below:

```
2022-01-13 23:32:18.7907|ERROR|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: The Game Euro Truck Simulator 2 didn't start for some reason (or the game uses a starter exe that launches the game itself)! so reverting changes back if needed...
2022-01-13 23:33:12.9838|ERROR|DisplayMagician.Program|WinLibrary/GetWindowsDisplayConfig: ERROR - DisplayConfigGetDeviceInfo returned WIN32STATUS ERROR_INVALID_PARAMETER when trying to query the adapter name for adapter 0.
2022-01-13 23:48:30.0180|WARN|DisplayMagician.Program|NVIDIALibrary/SetActiveConfig: There was an error disabling the display mode. NvAPI_Mosaic_EnableCurrentTopo() returned error code NVAPI_MODE_CHANGE_FAILED
```

The ERROR_INVALID_PARAMETER in DisplayConfigGetDeviceInfo line is a weird one. That line is when DM is trying to ask windows for the name of the adapter. The weird bit is that the adapter ID is 0. It is normally a number like 55768, 76223, or 45109 (windows chooses a new one each reboot). It shouldn't be 0. 0 is wrong. That's the reason that the adapter lookup fails. This seems like a NVIDIA or Windows issue to me.

I also noticed that your Surround setup has different adapter ids in the Windows DisplayConfigPaths as it does with the DisplayConfigModes. This shouldn't happen, and isn't a valid config. I wonder if this profile was set up when you trialed an early multi-threaded install for me, and that returned different adapter ids into the structure? Or its a bug with NVIDIA or Windows.

And lastly, you have the same error turning off Surround using NvAPI_Mosaic_EnableCurrentTopo() that @sk666 and @FormelLMS have been getting. Maybe it's all related to the above?

Can you please perform the following tests for me:

- Please install DMv2.2.0.50 for me: 2.2.0.50: [DisplayMagicianSetup-v2.2.0.50.zip](https://github.com/terrymacdonald/DisplayMagician/files/7886433/DisplayMagicianSetup-v2.2.0.50.zip)
- Please create a brand new Surround Profile and a brand new Three Screens profile.
- Please try to swap between those two new profiles and only those two new profiles.
- See if they work without any issue
- Please upload a support ZIP as well.

Thanks. 
Terry

**Comment by terrymacdonald** [comment]
Note: I updated the NVIDIALibrary and WnLibrary HDR logic as I was not sure they would always apply properly in Surround mode. They are now applying correctly on my test system. Will be interesting if that fixes the NVAPI_MODE_CHANGE_FAILED error (unlikely but possible)

This fix is in [DisplayMagicianSetup-v2.2.0.52.zip](https://github.com/terrymacdonald/DisplayMagician/files/7887117/DisplayMagicianSetup-v2.2.0.52.zip)

**Comment by erbkaiser** [comment]
I installed build 2.2.0.52 and recreated my profiles, then switched between the two a bit. Log attached. 
[DisplayMagician-Support-20220118-0947.zip](https://github.com/terrymacdonald/DisplayMagician/files/7887747/DisplayMagician-Support-20220118-0947.zip)

I did some additional testing, including re-editing all my game shortcuts to use the new display profiles and launched one of my truck sim games. So far, so good - I didn't encounter the beeping/delay problem at any of the display profile switches.

**Comment by terrymacdonald** [comment]
Hi @erbkaiser 

I've got another new version of DisplayMagician v2.2.0.112 ready to go.  This new test version of DisplayMagician v2.2.0.112 has a lot of new functionality:

- Is now a single instance, so games will start faster and display profiles will swap quicker
- Is now multithreaded to handle processing profile changes in the background
- Display layout detection is more reliable
- Now uses a new way of turning off NVIDIA surround so it avoids crashes on some video cards
- Stores the windows taskbar location and settings and attempts to reposition it to the correct place to fix NVIDIA driver issues
- Now allows you to uninstall the DisplayMagician Desktop Background Context Menu if you don't like it.

Please test it out, and tell me if there are any bugs. NOTE: There are quite a few changes in v2.2.0.112 compared to v2.1.5.5, so you will be asked to recreate new display profiles again. Please make sure you do this so that everything will work ok.
[DisplayMagicianSetup-v2.2.0.112.zip](https://github.com/terrymacdonald/DisplayMagician/files/7962890/DisplayMagicianSetup-v2.2.0.112.zip)

Thanks
Terry

**Comment by erbkaiser** [comment]
Hi Terry,

Good news for me and bad (I assume) news for you: I have completely changed my monitor setup this week and am now running one ultra wide 3440x1440 monitor over DisplayPort and a secondary 1920x1080 over HDMI. I'm still waiting on a monitor arm to be able to position the smaller monitor where I want it, so any tests with nVidia Surround and DisplayMagician will have to wait until that arrives. It also means I can no longer test with my absurd DP>DVI adapters.

**Comment by terrymacdonald** [comment]
Wow. Congratulations on getting such cool gear! Sounds like it's going to
be epic to use.

No problems on waiting for testing. I'm currently trying to figure out how
to force windows taskbars into their correct positions, and is not quite
working the way I'd hoped! I figure if the windows system settings
application can do it, then I should be able to reverse engineer how it was
done, and  then implement that myself. But so far no luck.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hey @erbkaiser,

Hope the year has been treating you well. I thought you might like to test the latest DisplayMagician v2.3.1.7 which is in the final stages of testing. It fixes a lot of issues since v2.2.0.112 that you had before, so I'd love it if you could give it a try. 

[DisplayMagicianSetup-v2.3.1.7.zip](https://github.com/terrymacdonald/DisplayMagician/files/8494367/DisplayMagicianSetup-v2.3.1.7.zip)

No pressure if you can't do any testing, just thought you'd like to get early notice before the release drops :).

Thanks
Terry

**Comment by erbkaiser** [comment]
Hey Terry,

great that you're still working on DM!

Unfortunately, I'm still in the middle of moving house so I don't have my PC setup. Currently working from a laptop with no external screen attached, so I won't be able to test until I have a desk space set up again.
